include/motor.h

```diff
@@ -176,20 +176,36 @@ class TextAPIItem {
 
     std::string set(std::string s) {
         std::string s2 = name_ + "=" + s;
-        char c[64];
+        char c[65];
         auto nbytes = motor_txt_->writeread(s2.c_str(), s2.size(), c, 64);
+        if (nbytes < 0) {
+            if (!no_throw_) {
+                throw std::runtime_error("text api set error " + std::to_string(nbytes) + ": " + strerror(-nbytes));
+            } else {
+                return "";
+            } 
+        }
         c[nbytes] = 0;
```

Comment:
Just FYI: you can make non-null terminated std::strings so you don't need to null terminate (or make it max-size +1 buffer) you could instead 
```
return {c, nbytes};
```
to use other other constructor. ([4] here: https://en.cppreference.com/w/cpp/string/basic_string/basic_string)